arraycontext/container/traversal.py

```diff
@@ -279,6 +291,52 @@ def rec(keys: Tuple[Union[str, int], ...],
 # }}}
 
 
+# {{{ array container reductions
+
+def rec_reduce_array_container(
```

Comment:
Seems to me like it is performing 2 operations at once i.e. map_and_reduce. How about we only provide a primitive for performing reduction over the children of a node. The user could call `rec_map_array_container` and then chain the call to `rec_reduce_array_container` over the mapped sub-arrays.

Reply:
Hm, that's a good point. I would prefer to keep it (mostly) as is because
* We don't have any use case at the moment for reduce only, right?
* This avoid traversing the container twice to do a mapreduce.

I'll rename the functions to `rec_map_reduce_array_container` to be clearer that the `array_func` thing can be more general. What do you think?

Reply:
93f6594

Reply:
Yep, I think the name change makes it explicit. And this is fine by me.